### DISCUSSION TITLE
Productivity improvements

### DISCUSSION THREAD

**Discussion by dca00** [texstudio-org/texstudio#3833] [Ideas]
Hi team,

For your new releases, could you please consider the following changes to user interactions:
A. For keyboard navigation, could you please change the current behavior as follows:
1. If the cursor is at the beginning of a word, and Ctrl-Shift-Right is pressed, select the word and the following blank space, until the beginning of the next word or punctuation mark.
2. If the cursor is at the beginning of a word, and Ctrl-Shift-Left is pressed, select the blank spaces to the left and the next word to the left.
3. If the cursor is at the end of a word, and Ctrl-Shift-Right is pressed, select the blank spaces to the right and the next word.
4. If the cursor is at the end of a word, and Ctrl-Shift-Left is pressed, select the word and the space to the left of it.

B. In the internal PDF viewer, could you flip the entire grid at a time, not 1 page at a time, or introduce a configuration item to choose one or the other behavior?

C. In the macro editor, please consider multiple selection in the list of macros, for deletion.

D. In the macro editor, please consider changing the tab order of the fields on the right side, to go through text boxes in the same order as they appear on the form.

E. In the macro editor, LaTeX content field, please consider using the main editor, with its code completion and syntax highlighting.

Any of the above will dramatically improve productivity.

Thanks for considering this!

**Comment by octaeder** [comment]
Ad B) do you use this option?

![grafik](https://github.com/user-attachments/assets/593aa607-8f9d-45ec-85d5-13137e8a0332)

If not, then all pages of the grid should be replaced by previous/next pages (integrated windowed pdf.viewer).

**Comment by octaeder** [comment]
Ad C) Multi select may also be of interest for moving Up or Down.

**Reply by octaeder** [reply]
simultaneous deleting of multiple macros has been implemented on October 10th.

**Comment by octaeder** [comment]
Ad D) I see following ordering:

![grafik](https://github.com/user-attachments/assets/609da344-ca90-42d7-b3c5-d2cde4156503)

Obviously cursor should not omit the shortcut field. But how could one include the description field? Since you are allowed to use tab characters in this field you won't be able to jump to the next field. This is the same for the code editor at the lower right.

**Comment by dca00** [comment]
Thanks for your tip re B) I was not aware it is a thing. However, a smart default for this should be off, to mimic Acrobat Reader's behavior. Since I never knew nor changed its value, and it is on, then I assume that this is the OOB default that is contrary to Acrobat; therefore, counter-intuitive.

For D) Here's something to put things into perspective: I use TS since you forked off TM. Since then, I've written 100s of manuscripts for the total of 10,000s of pages, on top of many short texts of which I have long lost count. My macro lists sometimes run 1-2 full screen heights, per manuscript. Never did I use the description field. Never had a need. And even if I pause to contemplate how I could have probably used it, were I to have a need for it, I cannot imagine ever using any rich formatting such as TAB.
Anyhow, presently, I have to tab from the name to the trigger and then click on Latex Content because 1) tab order seems to skip the latter most important text box and 2) there is no visual indication when it is focused. Never did I use any boxes other than name, trigger, and latex content. Never did I use shortcut. Never did I change the type. If the fields came in this exact order, and shortcut, description, and type followed latex content, this would have been the ideal layout for this form, and macro management for complex documents would not have been such a pain.

**Reply by octaeder** [reply]
@sunderme change default here `conf->registerOption("Preview/Single Page Step", &globalConfig->singlepagestep, true);`?

**Reply by sunderme** [reply]
I don't change defaults unless there is a compelling reason.
I don't see that here.

**Reply by dca00** [reply]
> I don't change defaults unless there is a compelling reason. I don't see that here.

Power to you!

It is like this: even if they do not say anything, still your users expect your PDF viewer to behave the same as the gold standard: Adobe Acrobat Reader.

If they behave the same, then your users are happy and appreciate your product.
If they behave differently, then they see it as an annoyance. Whichever floats your boat!

**Comment by dca00** [comment]
I'd like to add another request.
Next time you are in the code, could you make some improvement to the selection around direct speech?
Currently, Ctrl-Shift-Right arrow works ambiguously:
![image](https://github.com/user-attachments/assets/1c28a012-81bb-4974-911b-8ab35f0a7d2d)
![image](https://github.com/user-attachments/assets/637874af-2323-4e42-ac00-c8b726c68b73)
![image](https://github.com/user-attachments/assets/7a22551a-8172-4c0f-a866-6e77652f76d8)
If it could consistently stop on a word boundary, it would have been much more convenient.
